x-pack/plugins/observability_solution/synthetics/server/saved_objects/saved_objects.ts

```diff
@@ -52,33 +53,49 @@ export const registerUptimeSavedObjects = (
   encryptedSavedObjects.registerType(SYNTHETICS_SECRET_ENCRYPTED_TYPE);
 };
 
-export interface UMSavedObjectsAdapter {
-  config: UptimeConfig | null;
-  getUptimeDynamicSettings: UMSavedObjectsQueryFn<DynamicSettingsAttributes>;
-  setUptimeDynamicSettings: UMSavedObjectsQueryFn<void, DynamicSettingsAttributes>;
-}
-
-export const savedObjectsAdapter: UMSavedObjectsAdapter = {
-  config: null,
-  getUptimeDynamicSettings: async (client) => {
+export const savedObjectsAdapter = {
+  getSyntheticsDynamicSettings: async (
+    client: SavedObjectsClientContract
+  ): Promise<DynamicSettingsAttributes> => {
     try {
-      const obj = await client.get<DynamicSettingsAttributes>(settingsObjectType, settingsObjectId);
+      const obj = await client.get<DynamicSettingsAttributes>(
+        syntheticsSettingsObjectType,
+        syntheticsSettingsObjectId
+      );
       return obj?.attributes ?? DYNAMIC_SETTINGS_DEFAULT_ATTRIBUTES;
     } catch (getErr) {
-      const config = savedObjectsAdapter.config;
       if (SavedObjectsErrorHelpers.isNotFoundError(getErr)) {
-        if (config?.index) {
-          return { ...DYNAMIC_SETTINGS_DEFAULT_ATTRIBUTES, heartbeatIndices: config.index };
-        }
-        return DYNAMIC_SETTINGS_DEFAULT_ATTRIBUTES;
+        // If the object doesn't exist, check to see if uptime settings exist
+        return getUptimeDynamicSettings(client);
```

Comment:
I would remove this. It creates an confusing tie between Uptime and Synthetics in my opinion.

For example, the first time I used settings things were mirrored between Uptime and Synthetics. For the next uses, I expected that mirroring to continue, but it did not. 

The mirroring was due to this login where I initially had only an uptime setting SO and not a Synthetics one.

I would love to avoid falling back to Uptime, but I suppose we need this in place to account for the connection we currently have between the two? Is there another option? Some way to migrate the old Uptime settings into a new Synthetics SO once and then remove the need for this fallback?

Reply:
unfortunately there is no other way.